### PR TITLE
[Serializer] Deprecate SerializerAwareEncoder

### PR DESCRIPTION
**Pull request by JhonnyL**
| Q | A |
| --- | --- |
| Branch? | master |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | yes |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |

### CODE REVIEW

File: src/Symfony/Component/Serializer/Encoder/XmlEncoder.php [hunk 1]
```diff
@@ -21,8 +23,10 @@
  * @author Fabian Vogler <fabian@equivalence.ch>
  * @author Kévin Dunglas <dunglas@gmail.com>
  */
-class XmlEncoder extends SerializerAwareEncoder implements EncoderInterface, DecoderInterface, NormalizationAwareInterface
+class XmlEncoder implements EncoderInterface, DecoderInterface, NormalizationAwareInterface, SerializerAwareInterface
```
Comment (xabbuh) [inline src/Symfony/Component/Serializer/Encoder/XmlEncoder.php]: This is a BC break (in case someone type-hinted with `SerializerAwareEncoder`).

**Comment by JhonnyL** [comment]
@xabbuh I reverted the changes to XmlEncoder.

File: src/Symfony/Component/Serializer/Encoder/SerializerAwareEncoder.php [hunk 2]
```diff
@@ -11,23 +11,17 @@
...
-use Symfony\Component\Serializer\SerializerInterface;
 use Symfony\Component\Serializer\SerializerAwareInterface;
+use Symfony\Component\Serializer\SerializerAwareTrait;
 
 /**
  * SerializerAware Encoder implementation.
  *
  * @author Jordi Boggiano <j.boggiano@seld.be>
+ *
+ * @deprecated since version 3.1, to be removed in 4.0. Use the SerializerAwareTrait instead.
```
Comment (stof) [inline src/Symfony/Component/Serializer/Encoder/SerializerAwareEncoder.php]: the deprecation warning must be triggered too

Comment (stof) [inline src/Symfony/Component/Serializer/Encoder/SerializerAwareEncoder.php]: btw, this will be 3.2, not 3.1, as we are already feature-frozen

Comment (JhonnyL) [inline src/Symfony/Component/Serializer/Encoder/SerializerAwareEncoder.php]: But that would trigger a deprecation warning whenever XmlEncoder is used, with no option to fix it.

I will update to 3.2

Comment (xabbuh) [inline src/Symfony/Component/Serializer/Encoder/SerializerAwareEncoder.php]: @stof Isn't the docblock annotation enough as the rest is handled by the debug class loader?

**Comment by Ener-Getick** [comment]
LGTM :+1:

**Comment by xabbuh** [comment]
👍

**Comment by fabpot** [comment]
Thank you @JhonnyL.

**Pull request merged**